src/stack-pipe-workspace/stack-pipe.service.ts

```diff
@@ -151,14 +152,26 @@ export class StackPipeWorkspaceService {
     const errorList: string[] = [];
 
     // Check if there are any duplicate stack/pipe IDs.
-    const stackPipeIds = locations.map(sp => sp.stackPipeId).filter(Boolean);
-    stackPipeIds.forEach((stackPipeId, i) => {
-      if (stackPipeIds.findIndex(sp => sp === stackPipeId) !== i) {
-        errorList.push(
-          `[MONLOC-106-A] Duplicate Stack/Pipe found with ID: ${stackPipeId}`,
-        );
+    for (const location of locations) {
+      if (location.stackPipeId) {
+        const queryRunner = this.entityManager.connection.createQueryRunner();
+        await queryRunner.startTransaction();
+        const trx = queryRunner.manager;
+        const existingStackPipe = await this.getStackByNameAndFacId(
+          location.stackPipeId,
+          facId,
+          trx
+        ).catch(() => null);
+        if (existingStackPipe) {
+          const error = CheckCatalogService.formatResultMessage('MONLOC-106-A', {
+            fieldnames: 'stackPipeId',
+            recordtype: 'Stack Pipe'
+          });
+          errorList.push(error);
+          break;
+        }
       }
-    });
+    }
```

Comment:
This doesn't benefit from using a transaction, the `trx` parameter in `getStackByNameAndFacId` is optional, and the query runner isn't being released anyway. Also, I don't think catching the error and returning null is a good idea, should just let the error propagate.

However, I don't think this will solve the problem being seen. With these changes, this error will fire whenever a stack pipe is included in the import that already exists in the database, which can be valid in many cases.

```suggestion
    const stackPipeIds = locations.map(sp => sp.stackPipeId).filter(Boolean);
    stackPipeIds.forEach((stackPipeId, i) => {
      if (stackPipeIds.findIndex(sp => sp === stackPipeId) !== i) {
        errorList.push(
          `[MONLOC-106-A] Duplicate Stack/Pipe found with ID: ${stackPipeId}`,
        );
      }
    });
```